lib/ansible/modules/crypto/openssl_pkcs12.py

```diff
@@ -0,0 +1,261 @@
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: openssl_pkcs12
+author: "Guillaume Delpierre (@gdelpierre)"
+version_added: "2.4"
+short_description: Generate OpenSSL pkcs12 archive.
+description:
+    - "This module allows one to (re-)generate PKCS #12."
+requirements:
+    - "python-pyOpenSSL"
+options:
+    ca_certificates:
+        required: False
+        description:
+            - List of CA certificate to include.
+    certificate:
+        required: False
+        description:
+            - The path to read certificates and private keys from.
+              Must be in PEM format.
+    action:
+        required: False
+        default: 'export'
+        choices: [ 'parse', 'export' ]
+        description:
+            - Create (export) or parse a PKCS #12.
+    dest:
+        required: True
+        default: null
+        aliases: ['path', 'destfile']
+        description:
+            - Filename to write the PKCS#12 file to.
+    force:
+        required: False
+        default: False
+        choices: [ True, False ]
+        description:
+    friendly_name:
+        required: True
+        default: null
+        aliases: 'name'
+        description:
+            - Specifies the "friendly name" for the certificate and private key.
+    iter_size:
+        required: False
+        default: 2048
+        description:
+            - Number of times to repeat the encryption step.
+    maciter_size:
+        required: False
+        default: 1
+        description:
+            - Number of times to repeat the MAC step.
+    mode:
+        required: False
+        default: 0400
+        description:
+            - Default mode for the generated PKCS #12 file.
+    passphrase:
+        required: False
+        default: null
+        description:
+            - The PKCS#12 password.
+    privatekey:
+        required: True
+        description:
+            - File to read private key from.
+    privatekey_passphrase:
+        required: False
+        default: null
+        description:
+            - Passphrase source to decrypt any input private keys with.
+    state:
+        required: False
+        default: 'present'
+        choices: [ 'present', 'absent' ]
+        description:
+            - Whether the file should exist or not.
+'''
+
+EXAMPLES = '''
+name: 'Generate PKCS #12 file'
+openssl_pkcs12:
+  path: '/opt/certs/ansible.p12'
+  friendly_name: 'raclette'
+  privatekey_path: '/opt/certs/keys/key.pem'
+  cert_path: '/opt/certs/cert.pem'
+  ca_certificates: '/opt/certs/ca.pem'
+  state: present
+'''
+
+RETURN = '''
+'''
+
+import os
+
+try:
+    from OpenSSL import crypto
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.crypto import OpenSSLModule, OpenSSLModuleError, load_certificate, load_privatekey
+from ansible.module_utils.pycompat24 import get_exception
```

Comment:
There is no need to support Python 2.4, [see](https://github.com/ansible/ansible/blob/devel/docs/docsite/rst/roadmap/ROADMAP_2_4.rst#python-2-4-and-2-5-support-discontinuation).